scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -205,133 +197,111 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
         outputDirectory: File,
         downloadDirectory: File
     ): Map<Package, List<ScanResult>> {
-        val storageDispatcher = Executors.newFixedThreadPool(
-            NUM_STORAGE_THREADS,
-            NamedThreadFactory(ScanResultsStorage.storage.name)
-        ).asCoroutineDispatcher()
-
-        val scanDispatcher = Executors.newSingleThreadExecutor(NamedThreadFactory(scannerName)).asCoroutineDispatcher()
-
-        return try {
-            coroutineScope {
-                packages.withIndex().map { (index, pkg) ->
-                    val packageIndex = "(${index + 1} of ${packages.size})"
-
-                    async {
-                        pkg to scanPackage(
-                            pkg,
-                            packageIndex,
-                            downloadDirectory,
-                            outputDirectory,
-                            storageDispatcher,
-                            scanDispatcher
-                        )
-                    }
-                }.associate { it.await() }
+        val scannerCriteria = getScannerCriteria()
+
+        log.info { "Searching scan results for ${packages.size} packages." }
+
+        val remainingPackages = packages.filterTo(mutableListOf()) { pkg ->
+            !pkg.isMetaDataOnly.also {
+                if (it) this@LocalScanner.log.info { "Skipping '${pkg.id.toCoordinates()}' as it is meta data only." }
```

Comment:
Nit: You should be able to drop `this@` here.

Reply:
I wasn't sure about this, wouldn't `LocalScanner.log` always use the logger from the abstract `LocalScanner` class while `this@LocalScanner.log` uses the actual implementation like `ScanCode.log`?

Reply:
Nope, actually in both cases the log will show `INFO  org.ossreviewtoolkit.scanner.LocalScanner` (I just tried it).

Reply:
Thanks for checking, I'm a bit surprised. I have removed `this@`.